tests/playwright-test/test-step.spec.ts

```diff
@@ -0,0 +1,184 @@
+          for (const test of suite.tests) {
+            for (const result of test.results) {
+              for (const step of result.steps) {
+                console.log('%% ' + JSON.stringify(this.distillStep(step)));
+              }
+            }
+          }
+        };
+        processSuite(this.suite);
+      }
+    }
+    module.exports = Reporter;
+  `;
+
+  const result = await runInlineTest({
+    'reporter.ts': expectReporterJS,
+    'playwright.config.ts': `
+      module.exports = {
+        reporter: './reporter',
+      };
+    `,
+    'a.test.ts': `
+      const { test } = pwt;
+      test('pass', async ({ page }) => {
+        await test.step('outer step 1', async () => {
+          await test.step('inner step 1.1', async () => {});
+          await test.step('inner step 1.2', async () => {});
+        });
+        await test.step('outer step 2', async () => {
+          await test.step('inner step 2.1', async () => {});
+          await test.step('inner step 2.2', async () => {});
+        });
+      });
+    `
+  }, { reporter: '', workers: 1 });
+
+  expect(result.exitCode).toBe(0);
+  const objects = result.output.split('\n').filter(line => line.startsWith('%% ')).map(line => line.substring(3).trim()).filter(Boolean).map(line => JSON.parse(line));
+  expect(objects).toEqual([
+    {
+      category: 'hook',
+      title: 'Before Hooks',
+      steps: [
+        {
+          category: 'pw:api',
+          title: 'browserContext.newPage',
+        },
+      ],
+    },
+    {
+      category: 'test.step',
+      title: 'outer step 1',
+      steps: [
+        {
+          category: 'test.step',
+          title: 'inner step 1.1',
+        },
+        {
+          category: 'test.step',
+          title: 'inner step 1.2',
+        },
+      ],
+    },
+    {
+      category: 'test.step',
+      title: 'outer step 2',
+      steps: [
+        {
+          category: 'test.step',
+          title: 'inner step 2.1',
+        },
+        {
+          category: 'test.step',
+          title: 'inner step 2.2',
+        },
+      ],
+    },
+    {
+      category: 'hook',
+      title: 'After Hooks',
+      steps: [
+        {
+          category: 'pw:api',
+          title: 'browserContext.close',
+        },
+      ],
+    },
+  ]);
+});
+
+test('should report test.step from fixtures', async ({ runInlineTest }) => {
+  const expectReporterJS = `
+    class Reporter {
+      onStepBegin(test, result, step) {
+        console.log('%% begin ' + step.title);
+      }
+      onStepEnd(test, result, step) {
+        console.log('%% end ' + step.title);
+      }
+    }
+    module.exports = Reporter;
+  `;
+
+  const result = await runInlineTest({
+    'reporter.ts': expectReporterJS,
+    'playwright.config.ts': `
+      module.exports = {
+        reporter: './reporter',
+      };
+    `,
+    'a.test.ts': `
+      const test = pwt.test.extend({
+        foo: async ({}, use) => {
+          await pwt.test.step('setup foo', () => {});
```

Comment:
Does this work if the test object was extended again? How about if this was using the current `test` variable?